src/detection/PsfFormatter.cc

```diff
@@ -22,14 +22,13 @@
 #include "lsst/daf/persistence/BoostStorage.h"
 #include "lsst/daf/persistence/XmlStorage.h"
 #include "lsst/pex/exceptions.h"
-#include "lsst/pex/logging/Trace.h"
+#include "lsst/log/Log.h"
 #include "lsst/pex/policy/Policy.h"
 
 BOOST_CLASS_EXPORT(lsst::afw::detection::Psf)
 
-#define EXEC_TRACE  20
-static void execTrace(std::string s, int level = EXEC_TRACE) {
-    lsst::pex::logging::Trace("afw.detection.PsfFormatter", level, s);
+namespace {
+LOG_LOGGER _log = LOG_GET("afw.detection.PsfFormatter");
```

Comment:
@TallJimbo, the initialization code as it is written now depends on `log` being built as a shared object, in a shared object you can only call methods/functions after all globals in that shared object have been (dynamically) initialized. You are right to doubt that approach, it's not very portable and it may break if someone decides to link that code statically. It can be made safer of course, either with function-static approach or depending only on static initialization. Feel free to open a ticket if you think it needs improvement.